src/Stratis.Bitcoin.Features.PoA/BasePoAFeatureConsensusRules/PoAHeaderSignatureRule.cs

```diff
@@ -29,10 +41,28 @@ public override void Initialize()
             this.slotsManager = engine.SlotsManager;
             this.federationHistory = engine.FederationHistory;
             this.validator = engine.PoaHeaderValidator;
+
+            var lastCheckPoint = engine.Network.Checkpoints.LastOrDefault();
+            this.lastCheckPoint = (lastCheckPoint.Value != null) ? new HashHeightPair(lastCheckPoint.Value.Hash, lastCheckPoint.Key) : null;
         }
 
         public override async Task RunAsync(RuleContext context)
         {
+            // Only start validating at the last checkpoint block.
+            if (this.initialBlockDownloadState.IsInitialBlockDownload() && context.ValidationContext.ChainedHeaderToValidate.Height <= (this.lastCheckPoint?.Height ?? 0))
```

Comment:
I don't think we need to check IBD anymore if we are only checking the last checkpoint?

Reply:
IMO this `if` can be removed.

Reply:
It's the better of the two options when one considers that a `Guard.Assert(this.initialBlockDownloadState.IsInitialBlockDownload())` would be appropriate. We have to be sure that this is all happening in IBD as expected.